dataprofiler/profilers/utils.py

```diff
@@ -343,3 +343,63 @@ def biased_kurt(df_series):
 
     kurt = n * M4 / M2 ** 2 - 3
     return kurt
+
+def find_diff_of_numbers(stat1, stat2):
+    """
+    Finds the difference between two stats. If there is no difference, returns
+    "unchanged". For ints/floats, returns stat1 - stat2. For strings, returns
+    list containing [stat1, stat2]. For lists/sets, removes duplicates and
+    returns [unique values of stat1, shared values, unique values of stat2].
+
+    :param stat1: the first statistical input
+    :type stat1: Union[int, float, str, list, set]
+    :param stat2: the second statistical input
+    :type stat2: Union[int, float, str, list, set]
+    :return: the difference of the stats
+    """
+    diff = "unchanged"
+    if stat1 is None or stat2 is None:
+        diff = [stat1, stat2]
+    elif stat1 != stat2:
+        diff = stat1 - stat2
+    return diff
+
+
+def find_diff_of_strings(stat1, stat2):
+    """
+    Finds the difference between two stats. If there is no difference, returns
+    "unchanged". For strings, returns list containing [stat1, stat2]. 
+
+    :param stat1: the first statistical input
+    :type stat1: Union[str]
+    :param stat2: the second statistical input
+    :type stat2: Union[str]
+    :return: the difference of the stats
+    """
+    diff = "unchanged"
+    if stat1 != stat2:
+       diff = [stat1, stat2]
+    return diff
+
+
+def find_diff_of_lists_and_sets(stat1, stat2):
+    """
+    Finds the difference between two stats. If there is no difference, returns
+    "unchanged". Removes duplicates and returns [unique values of stat1, 
+    shared values, unique values of stat2].
+
+    :param stat1: the first statistical input
+    :type stat1: Union[list, set]
+    :param stat2: the second statistical input
+    :type stat2: Union[list, set]
+    :return: the difference of the stats
+    """
+    diff = "unchanged"
+    if stat1 is None or stat2 is None:
+        diff = [stat1, stat2]
+    elif set(stat1) != set(stat2):
+        unique1 = [element for element in stat1 if element not in stat2]
+        shared = [element for element in stat1 if element in stat2]
+        unique2 = [element for element in stat2 if element not in stat1]
+        diff = [unique1, shared, unique2]
```

Comment:
Should `unique1`, `shared`, `unique2` have types that reflect the types of `stat1` and `stat2`. i.e. if they are lists we return a list of lists, if they are sets we return a list of sets?

Reply:
I don't think it would add anything imo. I chose to make them interchangeable since I think we already have some lists/sets that interchange